avocado/plugins/resolvers.py

```diff
@@ -160,3 +162,34 @@ def resolve(self, reference):
         return ReferenceResolution(
             reference, ReferenceResolutionResult.SUCCESS, [runnable]
         )
+
+
+class RunnablesRecipeResolver(Resolver):
+    name = "runnables-recipe"
+    description = "Test resolver for multiple runnables in a JSON recipe file"
+
+    @staticmethod
+    def _validate_and_load_runnables(reference):
+        with open(reference, "r", encoding="utf-8") as json_file:
+            recipe = json.load(json_file)
+        if not validate_recipe_json_schema(recipe, "runnables-recipe.schema.json"):
```

Comment:
IMO, we don't need the schema validation here, because the runnable recipe schema is validated at line 185 in `Runnable.from_dict(d)`.

If we would just check if the recipe is a list, we can remove the whole `avocado/schemas/runnables-recipe.schema.json`. Because the only difference between `avocado/schemas/runnables-recipe.schema.json` and `avocado/schemas/runnable-recipe.schema.json is that the first one is a list of the second one. And the second one we are checking.

Reply:
Well, the most technically pure way of doing this is really using a JSON schema for the entire thing.  But, there are issues with the duplication of the schema that I'm not happy about, so I've substituted the check for:

```python
        with open(reference, "r", encoding="utf-8") as json_file:
            runnables = json.load(json_file)

        if not (
            isinstance(runnables, list) and all([isinstance(r, dict) for r in runnables])
        ):
```

That is, it should be a list with nothing but dicts inside.  The dicts themselves will be validated at `from_dict()`.

With this change, there's no external user for the validation code, so I've dropped the first commit.